aptos-move/block-executor/src/scheduler.rs

```diff
@@ -460,10 +466,9 @@ impl Scheduler {
         let mut validation_status = self.txn_status[txn_idx].1.write();
         self.set_aborted_status(txn_idx, incarnation);
 
-        // Schedule higher txns for validation, could skip txn_idx itself (needs to be
-        // re-executed first), but used to couple with the locked validation status -
-        // should never attempt to commit until validation status is updated.
-        if let Some(wave) = self.decrease_validation_idx(txn_idx) {
+        // Schedule higher txns for validation, skipping txn_idx itself (needs to be
+        // re-executed first).
+        if let Some(wave) = self.decrease_validation_idx(txn_idx + 1) {
```

Comment:
@danielxiangzl @sasha8 @zekun000 can you also think to make sure this change is fine? I think it should be.

Reply:
Can you provide context on why it was `txn_idx` before? Was this just an oversight?

Reply:
I do not see how this relates to rolling commit(?) I think this is fine.

Reply:
I don't understand this part, is it to skip the validation for txn_idx before execution? it may not increase the wave if the current idx == txn_idx + 1?

Reply:
In this case, it's the original behavior to reduce to txn_idx + 1, since txn_idx isn't ready for validation at that moment anyway. With rolling commit, we changed to reducing to txn_idx as it seemed easier to see the correctness that way. However, reducing to txn_idx + 1 should still work - the only time behavior is different is if current_idx = txn_idx + 1 as Zekun mentioned, in this case no need to start a new wave (so we also save wave update - maybe could lead to failing some CAS-es, invalidations, etc) as after the execution of txn_idx, either required_wave will be set, or it will be reset to txn_idx at that point and start a wave then.

Reply:
@danielxiangzl - I felt the same way before but now I think it's a bias of how we thought about the correctness. It's the same argument as without the rolling commit, where the fact that we have validation logic after finish_execution as well should save us? (and if we didn't, things would be broken without rolling commit as well?)

I think at that point I was more afraid of accidentally causing deadlocks with rolling commit logic, but now we have the code reduce_validation_index doesn't acquire any locks so it should be okay.